sway/commands/include_one.c

```diff
@@ -0,0 +1,78 @@
+#define _XOPEN_SOURCE 700 
+#include "sway/commands.h"
+#include "sway/config.h"
+#include "list.h"
+#include "log.h"
+#include "unistd.h"
+#include <string.h>
+#include <libgen.h>
+#include <wordexp.h>
+
+typedef struct {
+	char *name;
+	char *path;
+} config_location_map;
+
+int compare_paths(const config_location_map* item, char* abspath) {
+	if (strcmp(item->name, basename(abspath))==0) {
+		return 0;
+	}
+	return -1;
+}
+
+void priority_configs(const char *path, const char *parent_dir, struct sway_config *config, list_t *locations) {
+	char *wd = getcwd(NULL, 0);
+
+	if (chdir(parent_dir) < 0) {
+		sway_log(SWAY_ERROR, "failed to change working directory");
+		goto cleanup;
+	}
+
+	wordexp_t p;
+	if (wordexp(path, &p, 0) == 0) {
+		char **w = p.we_wordv;
+		size_t i;
+		for (i = 0; i < p.we_wordc; ++i) {
+			int index = list_seq_find(locations, (int (*)(const void *, const void *))compare_paths, w[i]);
+			char* matched_path = strdup(w[i]);
+			config_location_map config_loc = {basename(matched_path), matched_path};
+			if (index == -1) {
+				list_add(locations, &config_loc);
+				continue;
+			}
+			locations->items[index] = &config_loc; 
```

Comment:
config_loc is being allocated on the stack and a pointer to it being assigned to locations->items[index]

That really doesn't seem right.

Reply:
since config_loc is now a pointer you probably want to do locations->items[index] = config_loc. Otherwise you have a pointer to pointer (to the heap), so same issue and you are missing an deref somewhere.

Reply:
ohh I missed it. I will rectify it.